### PR TITLE
Uri Value Component

### LINKED ISSUE #17: Viewer Value Component: UriValueComponent
Implement `BaseValueComponent` for a `UriValue`.

### PR DESCRIPTION
**Pull request by mdelez**
closes #17

### CODE REVIEW

**Comment by tobiasschweizer** [comment]
Thanks, I try to have a look at that today.

**Comment by mdelez** [comment]
I think the validator could use some work but I'm not entirely sure what Knora accepts as a valid url

**Comment by tobiasschweizer** [comment]
> I think the validator could use some work but I'm not entirely sure what Knora accepts as a valid url

any valid `xsd:URI` (https://docs.knora.org/paradox/02-knora-ontologies/knora-base.html#subclasses-of-value, https://github.com/dasch-swiss/knora-api/blob/ccb64bbcf16b80d55423ff44b28c89d34bf39581/webapi/src/main/scala/org/knora/webapi/util/StringFormatter.scala#L713-L714). 

I think it would be good idea to put the validators in a central place so they could be reused. Also then the regex would only be created once and not with every instance of a value component.

**Comment by mdelez** [comment]
> any valid `xsd:URI` (https://docs.knora.org/paradox/02-knora-ontologies/knora-base.html#subclasses-of-value, https://github.com/dasch-swiss/knora-api/blob/ccb64bbcf16b80d55423ff44b28c89d34bf39581/webapi/src/main/scala/org/knora/webapi/util/StringFormatter.scala#L713-L714).
>
I'm a little lost on how I can use this. Do we have validation functions set up already for uri's or will we still need to use regex?

> I think it would be a good idea to put the validators in a central place so they could be reused. Also then the regex would only be created once and not with every instance of a value component.
>
I can work on this. It would probably be good to implement it before we create all of the other components. Should I make a new issue for this?

File: projects/knora-ui/src/lib/viewer/values/uri-value/uri-value.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+
+      expect(valueInputNativeElement.value).toEqual('http://www.google.ch');
+
+      expect(testHostComponent.inputValueComponent.form.valid).toBeFalsy();
+
+    });
+
+    it('should set a new display value', () => {
+
+      const newInt = new ReadUriValue();
```
Comment (tobiasschweizer) [inline projects/knora-ui/src/lib/viewer/values/uri-value/uri-value.component.spec.ts]: Could you adapt the variable name? Just for reasons of clarity, doesn't make any technical difference.

File: projects/knora-ui/src/lib/viewer/values/uri-value/uri-value.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+    it('should unsubscribe when destroyed', () => {
+      expect(testHostComponent.inputValueComponent.valueChangesSubscription.closed).toBeFalsy();
+
+      testHostComponent.inputValueComponent.ngOnDestroy();
+
+      expect(testHostComponent.inputValueComponent.valueChangesSubscription.closed).toBeTruthy();
+    });
+  });
+
+  describe('create an integer value', () => {
```
Comment (tobiasschweizer) [inline projects/knora-ui/src/lib/viewer/values/uri-value/uri-value.component.spec.ts]: please adapt the test's name

File: projects/knora-ui/src/lib/viewer/values/uri-value/uri-value.component.spec.ts (hunk 1 above)
Comment (mdelez) [reply projects/knora-ui/src/lib/viewer/values/uri-value/uri-value.component.spec.ts]: done in b228083

File: projects/knora-ui/src/lib/viewer/values/uri-value/uri-value.component.spec.ts (hunk 2 above)
Comment (mdelez) [reply projects/knora-ui/src/lib/viewer/values/uri-value/uri-value.component.spec.ts]: done in b228083

**Pull request merged**